### PR TITLE
Fix on upload with external storage 

### PR DESCRIPTION
**Pull request by claudiamurialdo**
Issue:91191
Given that UploadCachedFile keeps the absolute path of images that are uploaded on PrivateTempStorage of the external storage (so that it can distinguish between local disk images and external storage images), SetParameterMultimedia can receive values like these:

image_gxi = "GalaxySII.jpg" and image="gxupload:a31246ece3dd4899ae76599920bf5015"
where the image is translated to p.e. "https://genexus.blob.core.windows.net/myprivatecontainer/PrivateTempStorage/a39af8dd-8cf9-463b-a3bf-94aad44acbf4.tmp?sv=2018-03-28&sr=b&sig=a%2BwhZB6xJGAnfLUgQwmhKJy1YfQcHut3h1mwwlMgvzg%3D&st=2021-09-18T13%3A02%3A42Z&se=2021-09-19T13%3A02%3A42Z&sp=r"  which is an absolute URL. So it must make a relative URL to pass to the copy method (which expects an objectName)
Fix TryGetObjectNameFromURL to remove queryString (case of private files).

### CODE REVIEW

File: dotnet/src/dotnetframework/Providers/Storage/GXAzureStorage/AzureStorageExternalProvider.cs [hunk 1]
```diff
@@ -387,13 +387,13 @@ public bool TryGetObjectNameFromURL(string url, out string objectName)
 			string baseUrl = StorageUri + StorageUtils.DELIMITER + PublicContainer.Name + StorageUtils.DELIMITER;
 			if (url.StartsWith(baseUrl))
 			{
-				objectName = url.Replace(baseUrl, string.Empty);
+				objectName = StorageUtils.StripQueryString(url.Replace(baseUrl, string.Empty));
```
Comment (ggallotti) [inline dotnet/src/dotnetframework/Providers/Storage/GXAzureStorage/AzureStorageExternalProvider.cs:390]: I think this code should not be necessary, as it is in: [StorageFactory.cs]. Or I missing something. 
(https://github.com/genexuslabs/DotNetClasses/blob/923666b14cf531ca9a5541f8c6c59457e6c9e806/dotnet/src/dotnetframework/GxClasses/Storage/StorageFactory.cs).

```
public static bool TryGetProviderObjectName(ExternalProvider provider, string objectNameOrUrl, out string providerObjectName)
		{
			providerObjectName = null;
			if (provider != null && provider.TryGetObjectNameFromURL(objectNameOrUrl, out providerObjectName))
			{
				int idx = providerObjectName.IndexOf(QUESTION_MARK);
				if (idx > 0)
				{
					providerObjectName = providerObjectName.Substring(0, idx);
				}
				return true;
			}
			return false;
		}
```

**Review by ggallotti — DISMISSED** [review]

**Review by ggallotti — COMMENTED** [review]
Some Google Storage Test are failing. I don't know if this is random or is a side effect?

**Comment by claudiamurialdo** [comment]
> Some Google Storage Test are failing. I don't know if this is random or is a side effect?

I'll chek.

File: dotnet/src/dotnetframework/Providers/Storage/GXAzureStorage/AzureStorageExternalProvider.cs (hunk 1 above)
Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/Providers/Storage/GXAzureStorage/AzureStorageExternalProvider.cs]: Certainly. Replaced by StorageFactory.TryGetProviderObjectName.

**Review by ggallotti — APPROVED** [review]

**Pull request merged**